### ISSUE TITLE
Table in ":help compatible" needs updating

### ISSUE THREAD

**Issue by HaleTom** [vim/vim#1139]
In documenting the effect of `set nocompatible` on `&viminfo` (see Issue #1132), I noticed that the 49 variables currently listed in the [table](http://vimhelp.appspot.com/options.txt.html#%27compatible%27), are _all_ affected by `set compatible`.

If `nocompatible` resets `viminfo` to its default, shouldn't `compatible` set this to `NONE` or `""`?

**Comment by k-takata** [comment]
There are four types of options.
1. not affected by `:set cp`, not affected by `:set nocp`
2. set to vi default by `:set cp`, not affected by `:set nocp`
3. not affected by `:set cp`, set to vim default by `:set nocp`
4. set to vi default by `:set cp`, set to vim default by `:set nocp`

Maybe the table is incomplete.

**Comment by k-takata** [comment]
Here is a list of options affected by `:set cp` or `:set nocp`. (Extracted from options.c.)

```
    {"antialias",   "anti", P_BOOL|P_VI_DEF|P_VIM|P_RCLR,
    {"arabic",      "arab", P_BOOL|P_VI_DEF|P_VIM|P_CURSWANT,
    {"arabicshape", "arshape", P_BOOL|P_VI_DEF|P_VIM|P_RCLR,
    {"allowrevins", "ari",  P_BOOL|P_VI_DEF|P_VIM,
    {"backspace",   "bs",   P_STRING|P_VI_DEF|P_VIM|P_ONECOMMA|P_NODUP,
    {"backup",      "bk",   P_BOOL|P_VI_DEF|P_VIM,
    {"backupcopy",  "bkc",  P_STRING|P_VIM|P_ONECOMMA|P_NODUP,
    {"balloonexpr", "bexpr", P_STRING|P_ALLOCED|P_VI_DEF|P_VIM,
    {"breakindent",   "bri",  P_BOOL|P_VI_DEF|P_VIM|P_RWIN,
    {"cedit",       NULL,   P_STRING,
    {"cindent",     "cin",  P_BOOL|P_VI_DEF|P_VIM,
    {"compatible",  "cp",   P_BOOL|P_RALL,
    {"copyindent",  "ci",   P_BOOL|P_VI_DEF|P_VIM,
    {"cpoptions",   "cpo",  P_STRING|P_VIM|P_RALL|P_FLAGLIST,
    {"cscopepathcomp", "cspc", P_NUM|P_VI_DEF|P_VIM,
    {"cscoperelative", "csre", P_BOOL|P_VI_DEF|P_VIM,
    {"cscopetag",   "cst",  P_BOOL|P_VI_DEF|P_VIM,
    {"cscopetagorder", "csto", P_NUM|P_VI_DEF|P_VIM,
    {"cscopeverbose", "csverb", P_BOOL|P_VI_DEF|P_VIM,
    {"delcombine", "deco",  P_BOOL|P_VI_DEF|P_VIM,
    {"digraph",     "dg",   P_BOOL|P_VI_DEF|P_VIM,
    {"esckeys",     "ek",   P_BOOL|P_VIM,
    {"expandtab",   "et",   P_BOOL|P_VI_DEF|P_VIM,
    {"fileformats", "ffs",  P_STRING|P_VIM|P_ONECOMMA|P_NODUP,
    {"formatexpr", "fex",   P_STRING|P_ALLOCED|P_VI_DEF|P_VIM,
    {"formatoptions","fo",  P_STRING|P_ALLOCED|P_VIM|P_FLAGLIST,
    {"gdefault",    "gd",   P_BOOL|P_VI_DEF|P_VIM,
    {"history",     "hi",   P_NUM|P_VIM,
    {"hkmap",       "hk",   P_BOOL|P_VI_DEF|P_VIM,
    {"hkmapp",      "hkp",  P_BOOL|P_VI_DEF|P_VIM,
    {"hlsearch",    "hls",  P_BOOL|P_VI_DEF|P_VIM|P_RALL,
    {"incsearch",   "is",   P_BOOL|P_VI_DEF|P_VIM,
    {"indentexpr", "inde",  P_STRING|P_ALLOCED|P_VI_DEF|P_VIM,
    {"insertmode",  "im",   P_BOOL|P_VI_DEF|P_VIM,
    {"iskeyword",   "isk",  P_STRING|P_ALLOCED|P_VIM|P_COMMA|P_NODUP,
    {"joinspaces",  "js",   P_BOOL|P_VI_DEF|P_VIM,
    {"modeline",    "ml",   P_BOOL|P_VIM,
    {"more",        NULL,   P_BOOL|P_VIM,
    {"mzquantum",  "mzq",   P_NUM,
    {"numberwidth", "nuw",  P_NUM|P_RWIN|P_VIM,
    {"preserveindent", "pi", P_BOOL|P_VI_DEF|P_VIM,
    {"revins",      "ri",   P_BOOL|P_VI_DEF|P_VIM,
    {"ruler",       "ru",   P_BOOL|P_VI_DEF|P_VIM|P_RSTAT,
    {"scrolljump",  "sj",   P_NUM|P_VI_DEF|P_VIM,
    {"scrolloff",   "so",   P_NUM|P_VI_DEF|P_VIM|P_RALL,
    {"shelltemp",   "stmp", P_BOOL,
    {"shiftround",  "sr",   P_BOOL|P_VI_DEF|P_VIM,
    {"shortmess",   "shm",  P_STRING|P_VIM|P_FLAGLIST,
    {"showcmd",     "sc",   P_BOOL|P_VIM,
    {"showmode",    "smd",  P_BOOL|P_VIM,
    {"sidescrolloff", "siso", P_NUM|P_VI_DEF|P_VIM|P_RBUF,
    {"smartcase",   "scs",  P_BOOL|P_VI_DEF|P_VIM,
    {"smartindent", "si",   P_BOOL|P_VI_DEF|P_VIM,
    {"smarttab",    "sta",  P_BOOL|P_VI_DEF|P_VIM,
    {"softtabstop", "sts",  P_NUM|P_VI_DEF|P_VIM,
    {"startofline", "sol",  P_BOOL|P_VI_DEF|P_VIM,
    {"tagcase",     "tc",   P_STRING|P_VIM,
    {"tagrelative", "tr",   P_BOOL|P_VIM,
    {"termguicolors", "tgc", P_BOOL|P_VI_DEF|P_VIM|P_RCLR,
    {"textauto",    "ta",   P_BOOL|P_VIM,
    {"textwidth",   "tw",   P_NUM|P_VI_DEF|P_VIM|P_RBUF,
    {"tildeop",     "top",  P_BOOL|P_VI_DEF|P_VIM,
    {"ttimeout",    NULL,   P_BOOL|P_VI_DEF|P_VIM,
    {"undofile",    "udf",  P_BOOL|P_VI_DEF|P_VIM,
    {"viminfo",     "vi",   P_STRING|P_ONECOMMA|P_NODUP|P_SECURE,
    {"virtualedit", "ve",   P_STRING|P_ONECOMMA|P_NODUP|P_VI_DEF|P_VIM|P_CURSWANT,
    {"whichwrap",   "ww",   P_STRING|P_VIM|P_ONECOMMA|P_FLAGLIST,
    {"wildchar",    "wc",   P_NUM|P_VIM,
    {"writebackup", "wb",   P_BOOL|P_VI_DEF|P_VIM,
```

The behavior is controlled by the two flags `P_VI_DEF` and `P_VIM`.
1. `P_VI_DEF` (not in the list):  
   not affected by `:set cp`, not affected by `:set nocp`
2. `P_VI_DEF|P_VIM`:  
   set to vi default by `:set cp`, not affected by `:set nocp`
3. none:  
   not affected by `:set cp`, set to vim default by `:set nocp`
4. `P_VIM`:  
   set to vi default by `:set cp`, set to vim default by `:set nocp`

**Comment by HaleTom** [comment]
Thanks for the list and explanation. How about we change the `+` column to:

 `*` -  Reset to default by both
 `+` - Reset to default by `:set cp` only (I believe this follows the current documentation)
 `-` - Reset to default by `:set nocp` only

**Comment by HaleTom** [comment]
I wrote a [shell script](https://gist.github.com/HaleTom/a793106d85cf7be1bdb873631a9e1553) to update the flag column based upon:

 `*` -  Reset to default by both
 `+` - Reset to default by `:set cp` only (I believe this follows the current documentation)
 `-` - Reset to default by `:set nocp` only

[Here](https://github.com/HaleTom/vim/blob/compatible-doc/compatible/compatible_table) is the output. There are 21 new undocumented options (at the bottom).

I'm not a vim guru - could someone run through these and concisely summarise the effect of `set compatible` upon them (staying within 80 columns)?

Also, what do we say in the `effect` column when the character flag column is `-`?

**Comment by k-takata** [comment]
Fixed by #1161.
